packages/runtime/test/lib/testUtils.ts

```diff
@@ -132,6 +132,32 @@ export async function syncUntilHasIdentityDeletionProcess(transportServices: Tra
     return await syncUntilHas(transportServices, "identityDeletionProcesses", (m) => m.id === identityDeletionProcessId.toString());
 }
 
+export async function resetEventBusAndSyncUntilHasEvent<TEvent extends Event>(
+    runtimeServices: TestRuntimeServices,
+    subscriptionTarget: SubscriptionTarget<TEvent> & { namespace: string },
+    predicate?: (event: TEvent) => boolean
+): Promise<Event> {
+    runtimeServices.eventBus.reset();
+
+    let iterationNumber = 0;
+    let event: Event | undefined;
+    do {
+        await sleep(iterationNumber * 25);
+
+        await runtimeServices.transport.account.syncEverything();
+        event = runtimeServices.eventBus.publishedEvents.find(
+            (e) =>
+                e.namespace === subscriptionTarget.namespace &&
+                (typeof subscriptionTarget === "string" || e instanceof subscriptionTarget) &&
+                (!predicate || predicate(e as TEvent))
+        ) as TEvent | undefined;
+
+        iterationNumber++;
+    } while (!event && iterationNumber < 15);
```

Comment:
I have in mind that do-while loops aren't good style, although I can't seem to find the source I got that from (maybe it was a lecture in university). Therefore, I would rather refactor it to a while loop such that as soon as `event` is not undefined you return it. @jkoenig134 Do you know if that is a thing with avoiding do-while loops?

Also, my question remains why it is necessary to reset the event bus within the single tests.

Reply:
If you don't tell me whats wrong with do-while I will not vote for changing this. I don't have anything about it.

Reply:
It's not necessary to reset the event bus, I frequently do it to check that the events arrive at the right time

Reply:
The resets within the single tests are gone if that's what you meant